cve_bin_tool/csv2cve.py

```diff
@@ -87,17 +95,30 @@ def generate_modules(self):
                             f'Found CVES for {row["vendor"]} {row["product"]}, version {row["version"]}'
                         )
                         # if we found CVES add to the all_cves
-                        all_cves[row["product"]][row["version"]] = cves
+                        cve_data = dict()
+                        cve_data["vendor"] = row["vendor"]
+                        cve_data["product"] = row["product"]
+                        cve_data["version"] = row["version"]
+                        cve_data["cves"] = cves
+                        cve_data["cve_count"] = len(cve_data["cves"])
+
+                        if cve_data not in all_cves:
+                            all_cves.append(cve_data)
 
                     else:
                         self.logger.debug(
                             f'No CVEs found for {row["vendor"]} {row["product"]}, version {row["version"]}. Is the vendor/product info correct?'
                         )
                         # Vendor Product is wrong mark CVE_Number and Severity = UNKNOWN
-                        all_cves[row["product"]][row["version"]] = {
-                            "UNKNOWN": "UNKNOWN"
-                        }
-
+                        all_cves.append(
+                            {
+                                "vendor": row["vendor"],
+                                "product": row["product"],
+                                "version": row["version"],
+                                "cves": [{"cve_id": "UNKNOWN", "severity": "UNKNOWN"},],
```

Comment:
Why are we storing  dictionary of UNKNOWNs({"cve_id": "UNKNOWN", "severity": "UNKNOWN"}) in cves?